assembl/static2/js/app/components/search.jsx

```diff
@@ -130,26 +131,40 @@ const ImageType = (props) => {
   return <img className={props.className} src={`/static2/img/icon-${props.type}.svg`} alt="" />;
 };
 
+const getFieldAnyLang = (source, prop, locale) => {
+  let result;
+  if (!source) {
+    // Why does this happen?
+    return result;
+  }
+  if (source[`${prop}_${locale}`]) {
+    return source[`${prop}_${locale}`];
+  }
+  elasticsearchLangIndexes.forEach((lang) => {
+    const key = `${prop}_${lang}`;
+    if (source[key]) {
+      result = source[key];
+    }
+  });
+  if (result) {
+    return result;
+  }
+  if (source[`${prop}_other`]) {
+    return source[`${prop}_other`];
+  }
+  return result;
+};
+
 const DumbPostHit = (props) => {
   const locale = props.locale;
   const source = props.result._source;
-  const mtfromSuffix = locale === 'fr' ? 'en' : 'fr';
-  let subject = get(
-    props.result,
-    `highlight.subject_${locale}`,
-    get(props.result, `highlight.subject_${locale}-x-mtfrom-${mtfromSuffix}`, get(props.result, 'highlight.subject_und'))
-  );
+  let subject = getFieldAnyLang(props.result.highlight, 'subject', locale);
   if (!subject) {
-    subject = get(source, `subject_${locale}`, get(source, `subject_${locale}-x-mtfrom-${mtfromSuffix}`, source.subject_und));
+    subject = getFieldAnyLang(source, 'subject', locale);
   }
-  let body = get(
-    props.result,
-    `highlight.body_${locale}`,
-    get(props.result, `highlight.body_${locale}-x-mtfrom-${mtfromSuffix}`, get(props.result, 'highlight.body_und'))
-  );
+  let body = getFieldAnyLang(props.result.highlight, 'body', locale);
   if (!body) {
-    body = get(source, `body_${locale}`, get(source, `body_${locale}-x-mtfrom-${mtfromSuffix}`, source.body_und));
-    body = truncateText(body);
```

Comment:
Why did you remove `body = truncateText(body);`?

Reply:
silly error, correcting.